### ISSUE TITLE
Support for 8x16 grid?

### ISSUE THREAD

**Issue by fitzgen** [rbnstubbs/trellis-monome#2]
I glanced over the code and saw a few nested loops that had 8x8 hard coded in, so I assume that this only supports an 8x8 grid. Is that correct?

How much work would it take to optionally support 8x16?

**Comment by nw2s** [comment]
I saw this too... also just taking a look for integration with the nw2s::b. Looks like it might not be too much trouble. I'm getting my 128 kit in the mail today, so as soon as I get it running I'll be testing this and possibly making some merge requests.

**Comment by fitzgen** [comment]
Are you scottwilson from http://www.muffwiggler.com/forum/viewtopic.php?p=1675974 ? Because that was how I started looking into this :P

**Comment by nw2s** [comment]
Yes. This is a small world within a small world. The code is close. It looks like everything is configurable from within the ino file, but just needs a little clean up in the inner workings.

**Comment by rbnstubbs** [comment]
It would be very easy to add support for 8x16 grids, but the code may need some optimization in order to run an 8x16 smoothly - it still has some issues with an 8x8. Also if you are buying the kits you should know that this was written before they existed, and so it was made for an UNO, not a Leonardo. To change it over would be fairly simple, although I believe the libraries would need some editing. I just posted this on muffwiggler too

**Comment by fitzgen** [comment]
I haven't done very much embedded software at all, but it looks like there aren't really any proper profilers for Arduino :-/

Best thing I could find was this: http://www.dudley.nu/arduino_profiling/ but it requires manual instrumentation of the code. Might be nice to put all the instrumentation behind `#ifdef`s and actually check it in, so that it can be profiled more in the future.

I think I'm going to buy the 8x16 UNTZtrument kit and when it gets to me I'll work on profiling + porting over to Leonardo.

Aside: I have no idea what muffwiggler is (just stumbled upon that thread via google), but you guys seem into it. Want to give me the low down? Cool place?

**Comment by nw2s** [comment]
(taking it to email so we don't bug all the watchers!)

**Comment by nw2s** [comment]
Forked and built a new firmware that uses a lot of the original code, but cleans it up quite a bit. Uses a more recent version of Adafruit's library. I'm not a max user, though, so if anyone is and wants to try it, please see: 

https://github.com/nw2s/trellis-monome/blob/master/sketches/oontzmefirmware128/oontzmefirmware128.ino

**Comment by amj** [comment]
I'm not a max user either, so i'm naturally having some trouble getting it running with max ;)  Happy to make the code more 8x8 or 16x8 friendly -- would either of you like PRs or should i just make a fork of my own?

**Comment by rbnstubbs** [comment]
Oops haven't updated the project in a while... you're welcome to fork yourself!

The max portion was set up to be a fairly basic interpreter. Is it having trouble reading the arduino inputs?
